push_notification: Update the payload data that gets encrypted.

prakhar1144 opened the pull request:
<!-- Describe your pull request here.-->

Fixes: <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on api_docs/encrypted-payload-format.md:
```diff
@@ -0,0 +1,69 @@
+  "realm_name": "Zulip Dev",
+  "user_id": 10,
+  "sender_id": 6,
+  "recipient_type": "private",
```
Let's change this to `direct` to match `channel` with using the current name.

prakhar1144 commented:
Updates in the recent push:
* Addressed review comments
* [CZO comments](https://chat.zulip.org/#narrow/channel/378-api-design/topic/E2EE.20-.20encrypted.20APNs.20payload.20format/near/2236082):
  * dropped the `zulip_` prefix on `zulip_message_id` and `zulip_message_ids`
  * `message_ids` in the remove notification is a JSON list rather than a string joining numbers with `,`s

prakhar1144 commented on api_docs/encrypted-payload-format.md:
```diff
@@ -0,0 +1,69 @@
+  "content": "test content",
+  "sender_full_name": "aaron",
+  "sender_avatar_url": "https://secure.gravatar.com/avatar/818c212b9f8830dfef491b3f7da99a14?d=identicon&version=1"
+}
```
Done

timabbott commented on api_docs/changelog.md:
```diff
@@ -39,6 +39,10 @@ format used by the Zulip server that they are interacting with.
+* Mobile push notification payloads for FCM to remove push notifications
+  no longer contain the `zulip_message_id` field. Clients that use `zulip_message_ids`
+  have been out for years and we no longer support older client versions
+  that don't.
```
I put this all on a new API feature level, since it is nontrivial changes.

timabbott commented:
Looks great, did an API feature level bump and a bunch of documentation writing, including renaming the page, and am merging.
